### PR TITLE
algorithm: signum

### PR DESCRIPTION
**Pull request by SpiderMath**
[Open in Gitpod](https://gitpod.io/from-referrer/)&nbsp;[know more](https://www.gitpod.io/docs/pull-requests/)

### Describe your change:

- [x] Add an algorithm?
- [ ] Fix a bug or typo in an existing algorithm?
- [ ] Documentation change?

### Checklist:

- [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Javascript/blob/master/CONTRIBUTING.md).
- [x] This pull request is all my own work -- I have not plagiarized.
- [x] I know that pull requests will not be merged if they fail the automated tests.
- [x] This PR only changes one algorithm file. To ease review, please open separate PRs for separate algorithms.
- [x] All new JavaScript files are placed inside an existing directory.
- [x] All filenames should use the UpperCamelCase (PascalCase) style. There should be no spaces in filenames.
      **Example:**`UserProfile.js` is allowed but `userprofile.js`,`Userprofile.js`,`user-Profile.js`,`userProfile.js` are not
- [x] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
- [x] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

This PR is to add the implementation of the signum function. TheAlgorithms entry is [here](https://the-algorithms.com/algorithm/signum)

<a href="https://gitpod.io/#https://github.com/TheAlgorithms/JavaScript/pull/1266"></a>

### CODE REVIEW

**Review by appgurueu — COMMENTED** [review]
Looks fine; note: this will currently return `1` for NaN; returning NaN would be more sensible IMO.

**Comment by SpiderMath** [comment]
Oh yes. I completely missed that case. I'm sorry. Thanks for pointing it out, I'll fix it right away.

**Review by appgurueu — DISMISSED** [review]

File: Maths/Signum.js [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+  Wikipedia: https://en.wikipedia.org/wiki/Sign_function
+*/
+
+/**
+ * @param {Number} input
+ * @returns {-1 | 0 | 1 | NaN} sign of input (and NaN if the input is NaN)
+ */
+function signum (input) {
+  if (isNaN(input)) return NaN
+  else if (input === 0) return 0
```
Comment (raklaptudirm) [inline Maths/Signum.js:19]: Else is unnecessary as you are returning in each branch.

File: Maths/Signum.js [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+  else if (input < 0) return -1
+  else return 1
```
Comment (raklaptudirm) [inline Maths/Signum.js:21]: Still returns 1 for stuff like strings. I would suggest adding a conditional here and making NaN the default case.

**Review by raklaptudirm — CHANGES_REQUESTED** [review]

File: Maths/Signum.js (hunk 2 above)
Comment (SpiderMath) [reply Maths/Signum.js]: Fixed :+1:

File: Maths/Signum.js (hunk 1 above)
Comment (raklaptudirm) [reply Maths/Signum.js]: @SpiderMath looks like ya forgot about this.

Comment (SpiderMath) [reply Maths/Signum.js]: Oops

**Comment by appgurueu** [comment]
You still have an `else` in each branch. Generally using `else`-`if`s even if you are returning on each branch is not necessarily bad practice because it may help improve readability (so I'm fine with it either way), but in this particular example, your `if`s are simple enough that you should just get rid of the redundant `else`s as the `return`s already do the job.

**Review by appgurueu — APPROVED** [review]

**Review by raklaptudirm — APPROVED** [review]

**Pull request merged**